### DISCUSSION TITLE
First filer.sync maybe take a long time

### DISCUSSION THREAD

**Discussion by creeew** [seaweedfs/seaweedfs#3307] [Q&A] [answered]
ClusterB using filer.sync to sync clusterA's files,.
For the future file sync correctly we must use filer.sync to replay the clusterA's file to clusterB cause there's the only way metadata in cluserB will store clusterA's fid which called sourceFileId. If we just copy the clusterA's file to clusterB we will miss the sourceFileId. 

But what if clusterA is running a long time. There are a lot of meta log files in filer store. ClusterB maybe take a long time to sync all files in clusterA.

**What can I do to reduce the first filer.sync time cost?**

And **how can I know clusterB fully synced all files in clusterA**? (clusterA will be readonly when we first sync all files)
I was thinking retrieving all file's metadata in clusterB's filer store, caculating the file's md5 and comparing the file's md5 which caculated in clusterA. emmm sounds like a bad idea.

**Comment by chrislusf** [comment, marked as answer]
there is an effort to parallel the `filer.sync` https://github.com/chrislusf/seaweedfs/pull/3260

there are also metrics added for the `filer.sync` process.

**Reply by creeew** [reply]
That's good news. Looking forward to the upcoming feature.
Thanks for your advice!
